core/src/test/java/org/mule/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -904,6 +905,29 @@ public void testAll() throws MuleException, Exception
 
         assertEquals(1, threads);
     }
+    
+    @Test
+    public void testScatterGatherRouterWithCollectAllAndFlowVarsAggregationStrategy() throws MuleException, Exception
+    {
+        ScatterGatherRouter scatterGatherRouter = new ScatterGatherRouter();
```

Comment:
this tests initialises a scatter-gather and then doesn't stop it. Because it's manually creating the scatter-gather, it's also highly coupled to its internal structure and the assertion becomes hard to read.

Replace by a functional test